syntax = "proto3";

package server_db;

message Player {
	
	uint64 id = 1;
	string username = 2;
	string password = 3;
	uint32 wins = 4;
	uint32 tieds = 5;
	uint32 lost = 6;

}

message Game {
	
	uint64 id = 1;
	Player player1 = 2;
	Player player2 = 3;

	enum Result {
		TIED = 0;
		HAS_WINNER = 1;
	}

	Result result = 4;

	enum Winner {
		PLAYER1 = 0;
		PLAYER2 = 1;
	}

	Winner winner = 5;

}


message Request {

	enum Entity {
		PLAYER = 0;
		GAME = 1;
	}

	Entity entity = 1;

	enum Operation {

		CREATE = 0;
		RETRIEVE = 1;
		UPDATE = 2;
		DELETE = 3;

	}

	Operation operation = 2;


	Player player = 3;
	Game game = 4;
}



message Response {

	int32 status_code = 1;
	string message = 2;
}

message PlayerResponse {
	
	Response status = 1;
	repeated Player player = 2;
}

message GameResponse {

	Response status = 1;
	repeated Game game = 2;
}




